### PR TITLE
feat(hermes): add metrics

### PR DESCRIPTION
**Pull request by ali-bahjati**

### CODE REVIEW

File: hermes/src/config/rpc.rs [hunk 1]
```diff
@@ -13,9 +13,9 @@ const DEFAULT_RPC_REQUESTER_IP_HEADER_NAME: &str = "X-Forwarded-For";
 pub struct Options {
     /// Address and port the RPC server will bind to.
     #[arg(long = "rpc-listen-addr")]
-    #[arg(default_value = DEFAULT_RPC_ADDR)]
-    #[arg(env = "RPC_ADDR")]
-    pub addr: SocketAddr,
+    #[arg(default_value = DEFAULT_RPC_LISTEN_ADDR)]
+    #[arg(env = "RPC_LISTEN_ADDR")]
+    pub listen_addr: SocketAddr,
```
Comment (ali-bahjati) [inline hermes/src/config/rpc.rs:18]: these were not matching the cli args

File: hermes/Cargo.toml [hunk 2]
```diff
@@ -49,21 +49,6 @@ utoipa             = { version = "3.4.0", features = ["axum_extras"] }
...
-libp2p = { version = "0.42.2", features = [
-    "gossipsub",
-    "identify",
-    "mplex",
-    "noise",
-    "secp256k1",
-    "websocket",
-    "yamux",
-]}
-
```
Comment (ali-bahjati) [inline hermes/Cargo.toml:66]: Not used anymore

File: hermes/src/aggregate/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            if slot > *newest_observed_slot {
+                self.newest_observed_slot.insert(event.clone(), slot);
+                SlotOrder::New
+            } else {
+                SlotOrder::OutOfOrder
+            }
+        } else {
+            self.newest_observed_slot.insert(event.clone(), slot);
+            SlotOrder::New
+        };
```
Comment (Reisen) [inline hermes/src/aggregate/metrics.rs:99]:
```suggestion
        let order = if self.newest_observed_slot.get(&event)
            .map_or(true, |&observed_slot| slot > observed_slot) {
            self.newest_observed_slot.insert(event.clone(), slot);
            SlotOrder::New
        } else {
            SlotOrder::OutOfOrder
        };
```

**Review by Reisen — APPROVED** [review]
Very nice!

**Pull request merged**